src/app/store/effects/VehicleTechRecordModel.effects.ts

```diff
@@ -1,44 +1,56 @@
+  SetVehicleTechRecordModelVinOnCreateSucess,
+  UpdateVehicleTechRecord,
+  UpdateVehicleTechRecordSuccess
 } from '@app/store/actions/VehicleTechRecordModel.actions';
-import {Store} from '@ngrx/store';
-import {map, switchMap, tap, catchError} from 'rxjs/operators';
-import {forkJoin, Observable, of} from 'rxjs';
-import {Router} from '@angular/router';
-import {GetVehicleTestResultModel} from '../actions/VehicleTestResultModel.actions';
-import {TechnicalRecordService} from '@app/technical-record-search/technical-record.service';
-import {IVehicleTechRecordModelState} from '../state/VehicleTechRecordModel.state';
-import {SetErrorMessage, ClearErrorMessage} from '../actions/Error.actions';
-import {VEHICLE_TECH_RECORD_SEARCH_ERRORS} from '@app/app.enums';
+import { GetVehicleTestResultModel } from '../actions/VehicleTestResultModel.actions';
+import { TechnicalRecordService } from '@app/technical-record-search/technical-record.service';
+import { IVehicleTechRecordModelState } from '../state/VehicleTechRecordModel.state';
+import { SetErrorMessage, ClearErrorMessage } from '../actions/Error.actions';
+import { VEHICLE_TECH_RECORD_SEARCH_ERRORS } from '@app/app.enums';
+import { selectVehicleTechRecordModelHavingStatusAll } from '../selectors/VehicleTechRecordModel.selectors';
+import { VehicleTechRecordModel } from '@app/models/vehicle-tech-record.model';
+import { TechRecord } from '@app/models/tech-record.model';
+import {VehicleTechRecordUpdate} from '@app/models/vehicle-tech-record-update';
+import {SearchParams} from '@app/models/search-params';
 
 @Injectable()
 export class VehicleTechRecordModelEffects {
   @Effect()
   getTechnicalRecords$ = this._actions$.pipe(
     ofType<GetVehicleTechRecordModelHavingStatusAll>(EVehicleTechRecordModelActions.GetVehicleTechRecordModelHavingStatusAll),
     map(action => action.payload),
-    switchMap((payload) => this._technicalRecordService.getTechnicalRecordsAllStatuses(payload[0], payload[1]).pipe(
-      switchMap((techRecordJson: any) => of(new GetVehicleTechRecordModelHavingStatusAllSuccess(techRecordJson))),
-      tap((_) => {
-        this._store.dispatch(new GetVehicleTestResultModel(_.payload[0].systemNumber));
-        this._store.dispatch(new ClearErrorMessage());
-        this.router.navigate([`/technical-record`]);
+    switchMap((searchParams: SearchParams) => this._technicalRecordService.getTechnicalRecordsAllStatuses(searchParams.searchIdentifier, searchParams.searchCriteria).pipe(
+      switchMap((vTechRecords: VehicleTechRecordModel[]) => of(new GetVehicleTechRecordModelHavingStatusAllSuccess(vTechRecords))),
+      tap((action) => {
+        if (action.payload.length > 1 ) {
+          this.router.navigate(['/multiple-records']);
+        } else {
+          this._store.dispatch(new GetVehicleTestResultModel(action.payload[0].systemNumber));
+          this._store.dispatch(new ClearErrorMessage());
+          this.router.navigate(['/technical-record', 0]);
```

Comment:
dispatch another action here that will set the activeTechRecord. So we need an effect to pick this up and will then navigate to /technical-record. We would also need corresponding selector. Ideally we also need a selector that can be composable to give us non active techRecords for display later. Let discuss please. I will guide on this